op/model.py

```diff
@@ -423,6 +424,56 @@ def set_spec(self, spec, k8s_resources=None):
         self._backend.pod_spec_set(spec, k8s_resources)
 
 
+class StorageMapping(Mapping):
+    """Map of storage names to lists of Storage instances."""
+
+    def __init__(self, storage_names, backend):
+        self._backend = backend
+        self._storage_map = {storage_name: None for storage_name in storage_names}
+
+    def __contains__(self, key):
+        return key in self._storage_map
+
+    def __len__(self):
+        return len(self._storage_map)
+
+    def __iter__(self):
+        return iter(self._storage_map)
+
+    def __getitem__(self, storage_name):
+        storage_list = self._storage_map[storage_name]
+        if storage_list is None:
+            storage_list = self._storage_map[storage_name] = []
+            for storage_id in self._backend.storage_list(storage_name):
+                storage_list.append(Storage(storage_name, storage_id, self._backend))
+        return storage_list
+
+    def add(self, storage_name, count=1):
```

Comment:
I think Cory's _request_ name makes this looks more natural in the API since nothing will happen immediately after the method is called, and the storage may actually never show up depending. At the same time, the familiarity of _add_ is nice.  Perhaps let's use _request_ as the method name and document it as calling _storage-add_ so people grepping will find what they want.

Reply:
Agreed, done.